src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRulePipelineTests.cs

```diff
@@ -325,11 +367,93 @@ private void VerifyExecutionCount(CallbackActionService service, int expectedCou
             Assert.Equal(expectedCount, service.ExecutionTimestamps.Count);
         }
 
+        /// <summary>
+        /// Manually trigger for a number of iterations (<paramref name="iterationCount"/>) and test
+        /// that the actions are invoked for the number of expected iterations (<paramref name="expectedCount"/>) and
+        /// are throttled for the remaining number of iterations.
+        /// </summary>
+        private async Task ManualTriggerAsync(
+            ManualTriggerService triggerService,
+            CallbackActionService callbackService,
+            PipelineCallbacks callbacks,
+            int iterationCount,
+            int expectedCount,
+            MockSystemClock clock,
```

Comment:
Not necessary but would it make sense to register the clock using depdendency injection? Then in the test context register a mock clock?